meson.build

```diff
@@ -106,6 +106,8 @@ c_cpp_args = [
   # We use absolute include paths as much as possible.
   '-I' + meson.project_source_root(),
   '-I' + meson.project_build_root(),
+  # Allow detection of Meson for migration purposes.
```

Comment:
nit: a "FIXME" like you had with the bazel vendoring changes might make this easier to grep for when we migrate to bazel fully

Reply:
I mean. It'll get deleted when we do `rm -rf **/meson.build`. So I don't think it's strictly necessary.